protostar/protostar_cli.py

```diff
@@ -174,28 +197,31 @@ def _check_git_version(self):
             )
 
     async def run(self, args: Any) -> None:
-        logger = self._setup_logger(args.no_color)
+        self._setup_logger(args.no_color)
         has_failed = False
-
         try:
+            await self.upgrade_checker.check_for_upgrades_if_necessary()
```

Comment:
```suggestion
            await self.upgrade_checker.check_for_upgrades()
```
I think this postfix is not very informative.

Reply:
Wouldn't be better to make this check after the command? First, user wouldn't have to wait for it before he sees what he wants, second it won't dissapear under command output. I think `pip` and `gh` print notifications at the bottom.

Reply:
Outvoted. I'll remove it.

Reply:
If it's a convention it makes kinda sense to name it this way. It is weird because technically you shouldn't write methods like this. If you need to include such postfix you break single respoinsibility rule 90% of the time.

What do you think about writing it like this:

self.upgrade_checker.load_chached_info() or (await self.upgrade_checker.load_remote_info() )
self.log_new_version_info()

this way you make stuff explicit and don't hide log underneath check.
You can add assert in log to make it secure.

Reply:
That's better. I'll split the method.

Reply:
It's a bit more complicated than that. I hide ifs in the `run` method. `run` is less explicit but not confusing.